### DISCUSSION TITLE
Windows环境下的配置和环境准备

### DISCUSSION THREAD

**Discussion by zRzRzRzRzRzRzR** [zai-org/ChatGLM3#250] [Q&A]
在这里提出Windows环境下配置ChatGLM3-6B模型的任何问题，例如：
+ Windows下的cuda和torch跟Linux要求一样么
+ Windows 7 系统能运行这个仓库么

**Comment by zhangyanming** [comment]
请问，Windows 下如何更换成已下载的ChatGLM3-6B-32K？

**Comment by qmd158** [comment]
下载了modelScope/chatglm3-demo模型到本地，
环境变量也配置了，
项目拉取得是ChatGLM3
但是执行streamlit run main.py
报错：ValueError: The current device_map had weights offloaded to the
disk. Please provide an offload_folder for them. Alternatively, make
sure you have safetensors installed if the model you are using
offers the weights in this format.
如何解决？有没有大佬遇到过？拉取得ChatGLM3项目需不需要改项目代码？

**Reply by qmd158** [reply]
我的电脑是tinkpad E15

**Comment by wenrui123123** [comment]
运行所有demo都，与AI互动的时候都出现RuntimeError: "addmm_impl_cpu_" not implemented for 'Half'这个错误，大佬们知道怎么解决吗

**Reply by ascode** [reply]
我也是，新手，有点懵逼

**Reply by zRzRzRzRzRzRzR** [reply]
这个大概率是你用cpu了吧

**Comment by LucaLucasX** [comment]
部署模型时怎么指定路径呢

**Reply by zRzRzRzRzRzRzR** [reply]
使用绝对路径

**Comment by featherfly0513** [comment]
我按照步骤操作到安装依赖环境pip install -r requirements.txt，在vllm这一项一直报错，搜索了之后才知道vllm用在Linux上，我是win10的系统，有没有人指导下后面怎么操作。
![企业微信截图_171807295980](https://github.com/THUDM/ChatGLM3/assets/33362813/29826b62-fb21-4a11-84fd-fd95e8e8e964)

**Reply by lookatall-to** [reply]
同样错误，我的是windows10 专业版

**Reply by 18534902089** [reply]
How can we solve this, guys?

**Reply by hzucc** [reply]
guys +1。快说，怎么解决，不说的话我就投票给特朗普。一查浏览器就看到一堆win10部署这个模型的帖子，我都感觉我是在平行世界了。有没有官方人员帮帮忙。I am a new hand ， small white。

**Reply by badass-16** [reply]
遇到同上问题+1，开发者怎么都默认是在Linux环境下玩？我在Win11环境下按官网或者相关参考书籍的教程走下来，不是缺.DLL就是报字典无法定位（vocabulary not found）的错误，没谁了。

**Reply by flyingAfish** [reply]
把vllm>=0.4.2从requirements.txt里面去掉就阔以啦

**Comment by lookatall-to** [comment]
您好，邮件已收到，谢谢！

**Comment by lookatall-to** [comment]
您好，邮件已收到，谢谢！
